runtime/include/chpl-tasks.h

```diff
@@ -243,15 +243,38 @@ char* chpl_task_idToString(
                chpl_taskID_t); //Task ID
 
 //
-// Yield.
+// Yields the current task. If Quiescent State-Based Reclamation
+// is enabled, it will invoke a checkpoint which will invalidate
+// any references to any protected data.
 //
 void chpl_task_yield(void);
 
+//
+// Yields the current task. If Quiescent State-Based Reclamation
+// is enabled, it will invoke a checkpoint if the passed counter
+// is some multiple of CHPL_QSBR_ITERATIONS_PER_CHECKPOINT. This
+// should be invoked during tight loops which yield repeatedly.
+//
+void chpl_task_yield2(int);
+
 //
 // Suspend.
 //
 void chpl_task_sleep(double);
 
+//
+// (Optional) Invoked by a thread that may be idle for an
```

Comment:
Do you really mean "thread" here, or do you mean "task"?  I think the latter.  If so, the comment should be updated.

Reply:
Nope, it needs to be called from an idle thread. Its what gets called from the scheduler.

Reply:
Okay.  But it's not necessarily "idle", is it?  It just doesn't have Chapel work to do.  (Because not only will we call this from threads looking for work in the tasking layer thread schedulers but also from user pthreads that are going to go off and do non-Chapel things.)  Nevertheless, as in the `yield2` case, given the current use of this I'm not sure what the right term would be.  For now, though, can we at least rename this as `chpl_task_thread_onPark()`?  I want to get "thread" into the name because although we've gone to some effort to hide the fact that there are such things as threads that host Chapel tasks, in some cases (as here) that shows through anyway and when it does we want to be explicit about it.